src/overlays/actors/ovl_En_Ru1/z_en_ru1.c

```diff
@@ -1444,7 +1444,7 @@ void func_80AEDB30(EnRu1* this, PlayState* play) {
     }
 }
 
-void func_80AEDEF4(EnRu1* this, PlayState* play) {
+void EnRu1_UpdateSpeedXZ(EnRu1* this, PlayState* play) {
     f32* speedXZ = &this->actor.speed;
     DynaPolyActor* dynaPolyActor = DynaPoly_GetActor(&play->colCtx, this->actor.floorBgId);
 
```

Comment:
This function does 3 things:
- If ruto is standing on a treasure chest, it gives her some speed to make her walk off of it (presumably)
and 
- Clamps or otherwise limits speed when using a debug feature, `kREG(27)`. 
- otherwise, sets speed to 0

I think calling this "update speed" is a little too broad, but I unfortunately dont have any better suggestions at the moment

Reply:
sorry the condition involving `kREG(27)` still seems relevant in non-debug contexts. But you can assume that value is 0, because it never gets set in practice. So it seems like an overall clamp on her speed.